src/v/coproc/api.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "coproc/event_listener.h"
+#include "coproc/pacemaker.h"
+#include "coproc/sys_refs.h"
+
+#include <seastar/core/coroutine.hh>
+
+#include <vector>
+
+namespace coproc {
+
+class api {
+public:
```

Comment:
Please use a separate implementation file, and minimise includes here, e.g., `sharded` and `unique_ptr` don't require a complete type.

Otherwise a lot of coproc is brought into application.h, which has a large knock on effect for recompile times.